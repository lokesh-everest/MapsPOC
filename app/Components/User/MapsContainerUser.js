import React, {Component} from 'react'
import MapsUser from "./MapsUser";
import SocketIOClient from 'socket.io-client';

export default class MapsContainerUser extends Component {
    constructor(props) {
        super(props);
        const sourceCord = {
            latitude: 12.974963,
            longitude: 77.609139
        };
        const destCord = {
            latitude: 12.970400,
            longitude: 77.637398
        };
        this.socket = SocketIOClient('http://13.233.90.8:3000');
        this.state = {
            currentSourceCord: sourceCord,
            destCord: destCord,
            markers: [{
                key: 1,
                coordinates: sourceCord,
                title: "Source",
                imagePath: require("./../../assets/restaurant.png")
            }, {
                key: 2,
                coordinates: destCord,
                title: "Destination",
                imagePath: require("./../../assets/home.png")
            }],
            initialMap: {
                latitude: sourceCord.latitude,
                longitude: sourceCord.longitude,
                latitudeDelta: 0.0922,
                longitudeDelta: 0.0421
            },
            driverCoordinates: sourceCord,
            traveledPathCoordinates: []
        }
    }

    componentDidMount() {
        this.socket.on('customerEvent', (message) => {
            this.setState({driverCoordinates:message});
        });
    }

    render() {
        return (
            <MapsUser testId='map' markers={this.state.markers} driverCoordinates={this.state.driverCoordinates}
                      initalMap={this.state.initialMap}
                      sourceCoordinates={this.state.currentSourceCord} destinationCoordinates={this.state.destCord}/>
        )
    }

}