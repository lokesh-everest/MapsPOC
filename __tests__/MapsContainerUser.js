import 'jsdom-global/register';
import 'react-native';
import React from 'react';
import MapsContainerUser from "../app/Components/User/MapsContainerUser";

import renderer from 'react-test-renderer';
import { shallow,mount,configure } from 'enzyme';

describe(MapsContainerUser,()=>{
    it('render correcly',()=>{
        const wrapper = shallow(<MapsContainerUser />);
        expect(wrapper).toMatchSnapshot();
    })
})
