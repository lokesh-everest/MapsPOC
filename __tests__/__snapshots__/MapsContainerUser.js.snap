// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapsContainerUser render correcly 1`] = `
<MapsUser
  destinationCoordinates={
    Object {
      "latitude": 12.9704,
      "longitude": 77.637398,
    }
  }
  driverCoordinates={
    Object {
      "latitude": 12.974963,
      "longitude": 77.609139,
    }
  }
  initalMap={
    Object {
      "latitude": 12.974963,
      "latitudeDelta": 0.0922,
      "longitude": 77.609139,
      "longitudeDelta": 0.0421,
    }
  }
  markers={
    Array [
      Object {
        "coordinates": Object {
          "latitude": 12.974963,
          "longitude": 77.609139,
        },
        "imagePath": Object {
          "testUri": "../../../app/assets/restaurant.png",
        },
        "key": 1,
        "title": "Source",
      },
      Object {
        "coordinates": Object {
          "latitude": 12.9704,
          "longitude": 77.637398,
        },
        "imagePath": Object {
          "testUri": "../../../app/assets/home.png",
        },
        "key": 2,
        "title": "Destination",
      },
    ]
  }
  sourceCoordinates={
    Object {
      "latitude": 12.974963,
      "longitude": 77.609139,
    }
  }
  testId="map"
/>
`;
